.github/workflows/build_and_test.yml

```diff
@@ -367,6 +367,17 @@ jobs:
     steps:
     - name: Checkout Spark repository
       uses: actions/checkout@v2
+    - name: Cache Scala, SBT and Maven
```

Comment:
Nit: is there any way to avoid duplicated specs for this step "Cache Scala, SBT and Maven"

Reply:
I tired to find a way so hard but couldn't find so far.

Reply:
OK that's fine :)

Reply:
It seems there are many requests for that, but GA still does not support it: https://github.community/t/support-for-yaml-anchors/16128